src/accelerate/tracking.py

```diff
@@ -151,7 +153,9 @@ def store_init_configuration(self, values: dict):
         """
         self.writer.add_hparams(values, metric_dict={})
         self.writer.flush()
-        logger.info("Stored initial configuration hyperparameters to TensorBoard")
+        with open("self.logging_dir/hparams.yaml", "w") as outfile:
+            yaml.dump(values, outfile)
```

Comment:
Do we need to check if all keys/vals are serializable here, or are we going to let the error yaml throws be the one we present to users in this case? (not sure if this is done elsewhere in the codebase)

Reply:
We do that somewhat here: https://github.com/huggingface/accelerate/blob/main/src/accelerate/tracking.py#L170-L176

Maybe a try/catch that alerts the user that non-serializable bits were tried to be stored? Not sure what the try/catch error would be for that, otherwise I'm open to it just throwing a yaml error if we cannot :)

Reply:
could catch a `yaml.representer.RepresenterError` and return the message maybe?

Reply:
Along with more descriptive explanation of what happened under the hood accelerate...otherwise its the same as letting yaml handle it. 

I'm fine with either solution, just wanted to bring it up!

Reply:
Here, I think we can use`safe_dump` instead of  `dump` as it serialises only the basic tags and then we can add a try/catch with yaml error.

Reply:
I have added the try/catch error.

Reply:
I am not sure where to give a descriptive explanation. Should I add it as a comment or should I define it in the PR?

Reply:
Doing so in the `logger.error` (see my suggestions) is good enough :)